internal/controller/pipelinerollout/pipelinerollout_controller_test.go

```diff
@@ -659,10 +659,10 @@ func Test_processExistingPipeline_PPND(t *testing.T) {
 			initialInProgressStrategy:      apiv1.UpgradeStrategyPPND,
 			numaflowControllerPauseRequest: &falseValue,
 			isbServicePauseRequest:         &falseValue,
-			expectedInProgressStrategy:     apiv1.UpgradeStrategyNoOp,
+			expectedInProgressStrategy:     apiv1.UpgradeStrategyPPND,
 			expectedRolloutPhase:           apiv1.PhaseDeployed,
 			expectedPipelineSpecResult: func(spec numaflowv1.PipelineSpec) bool {
-				return reflect.DeepEqual(ctlrcommon.PipelineWithDesiredPhase(pipelineSpecWithTopologyChange, numaflowv1.PipelinePhaseRunning), spec)
+				return reflect.DeepEqual(ctlrcommon.PipelineWithDesiredPhase(pipelineSpecWithTopologyChange, numaflowv1.PipelinePhasePaused), spec)
```

Comment:
What's happening here? Previously, it seems that if the spec were already applied, then we would unpause the pipeline and take it out of UpgradeInProgress=PPND. So, what's happening differently now?

Reply:
Can you investigate further? I know this unit test is a little tough since sometimes things take multiple reconciliations in certain cases. But we need to understand what changed that is causing this.

Reply:
I think the metadata must need updating if the Pipeline is staying paused.

Reply:
> Can you investigate further? I know this unit test is a little tough since sometimes things take multiple reconciliations in certain cases. But we need to understand what changed that is causing this.

can do!

Reply:
This is being caused by us checking the annotations and labels in `ChildNeedsUpdating` in the PipelineRollout reconciler. When commenting those checks out, this test case works with the expected values it did before.

I also tried reconciling the rollout multiple times and that did not make the test pass.

Reply:
Potentially found and fixed the issue - the existingPipelineDef doesn't have the labels set that are added to Pipeline when they are created with PPND.

Tested this locally by creating PipelineRollout and checking labels of initial Pipeline.

Reply:
Issue was resolved as we found that using `DeepEqual` for maps will not consider nil maps equal to empty maps and thus we created a helper function that checks this first.